### PR TITLE
Catkin plugin: Handle rosdep empty and multiple dependencies.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR resolves both LP: [#1548406](https://bugs.launchpad.net/snapcraft/+bug/1548406) and [#1548404](https://bugs.launchpad.net/snapcraft/+bug/1548404) by refactoring the `rosdep` dependency resolution code to account for an empty result as well as multiple results on one line.

Note that both of these bugs are fixed here since, while they're separate bugs, the fix for both is very closely related and it would be more work to split them apart than to just do both.

### CODE REVIEW

File: snapcraft/plugins/catkin.py [hunk 1]
```diff
@@ -350,9 +350,9 @@ def _find_system_dependencies(catkin_packages, ros_distro, ros_package_path,
...
             # but the developer could have also forgotten to tell us to build
             # it.
-            system_dependency = rosdep.resolve_dependency(dependency)
-
-            if not system_dependency:
+            try:
+                these_dependencies = rosdep.resolve_dependency(dependency)
+            except SystemDependencyNotFound:
                 raise RuntimeError(
                     'Package "{}" isn\'t a valid system dependency. '
```
Comment (sergiusens) [inline snapcraft/plugins/catkin.py]: s/"{}"/{!r}/

File: snapcraft/plugins/catkin.py [hunk 2]
```diff
@@ -446,20 +451,16 @@ def resolve_dependency(self, dependency_name):
             output = self._run(['resolve', dependency_name, '--rosdistro',
                                 self._ros_distro, '--os', 'ubuntu:trusty'])
         except subprocess.CalledProcessError:
-            return None
+            raise SystemDependencyNotFound(
+                '"{}" does not resolve to a system dependency'.format(
```
Comment (sergiusens) [inline snapcraft/plugins/catkin.py]: s/"{}"/{!r}/

File: snapcraft/plugins/catkin.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/plugins/catkin.py]: Done.

File: snapcraft/plugins/catkin.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/plugins/catkin.py]: Done.

**Comment by sergiusens** [comment]
:+1:

thanks!

**Pull request merged**